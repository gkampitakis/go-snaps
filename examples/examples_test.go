package examples

import (
	"flag"
	"fmt"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

// You can use -update flag to control if you want to update the snapshots
// go test ./... -v -update
var updateSnapshot = flag.Bool("update", false, "update snapshots flag")

func TestUpdateWithFlag(t *testing.T) {
	snaps := snaps.WithConfig(snaps.Update(*updateSnapshot))

	inputs := []string{
		"lore ipsum dolor sit amet",
		"consectetur adipiscing elit",
		"sed do eiusmod tempor incididunt ut labore et dolore magna aliqua",
		"Ut enim ad minim veniam, quis nostrud laboris nisi ut aliquip ex ea commodo consequat.",
	}

	for i, input := range inputs {
		t.Run(fmt.Sprintf("test - %d", i), func(t *testing.T) {
			snaps.MatchSnapshot(t, input)
		})
	}
}
