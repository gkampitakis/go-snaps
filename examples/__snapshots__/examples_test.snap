
[TestUpdateWithFlag/test_-_0 - 1]
lore ipsum dolor sit amet
---

[TestUpdateWithFlag/test_-_1 - 1]
consectetur adipiscing elit
---

[TestUpdateWithFlag/test_-_2 - 1]
sed do eiusmod tempor incididunt ut labore et dolore magna aliqua
---

[TestUpdateWithFlag/test_-_3 - 1]
Ut enim ad minim veniam, quis nostrud laboris nisi ut aliquip ex ea commodo consequat.
---
